relayer/rpc/http-client.ts

```diff
@@ -0,0 +1,122 @@
+import { setTimeout } from "timers/promises";
+
+/**
+ * A simple HTTP client with exponential backoff retries and 429 handling.
+ */
+export class HttpClient {
+  private initialDelay: number = 1_000;
+  private maxDelay: number = 60_000;
+  private retries: number = 0;
+  private timeout: number = 5_000;
+
+  constructor(options?: HttpClientOptions) {
+    options?.initialDelay && (this.initialDelay = options.initialDelay);
+    options?.maxDelay && (this.maxDelay = options.maxDelay);
+    options?.retries && (this.retries = options.retries);
+    options?.timeout && (this.timeout = options.timeout);
+  }
+
+  public async get<T>(url: string, opts?: HttpClientOptions): Promise<T> {
+    return this.executeWithRetry(url, "GET", opts);
+  }
+
+  private async execute<T>(
+    url: string,
+    method: string,
+    opts?: HttpClientOptions,
+  ): Promise<T> {
+    let response;
+    try {
+      response = await fetch(url, {
```

Comment:
is this just globally available? :thinking: why is that?

Reply:
good question, we are already using it in source-tx middleware, thats why I used it here as well. We dont have an engine version specified, but this would mean we need node v18+ right? I could use somehting else if we want to go lower.

Reply:
v18 is LTS. I'm fine with keeping as is. `fetch` is `stable`.